packages/console/src/pages/ApplicationDetails/components/Permissions/index.tsx

```diff
@@ -0,0 +1,91 @@
+import { type Application, type ApplicationUserConsentScopesResponse } from '@logto/schemas';
+import { useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import useSWR from 'swr';
+
+import ActionsButton from '@/components/ActionsButton';
+import Breakable from '@/components/Breakable';
+import FormCard from '@/components/FormCard';
+import TemplateTable from '@/components/TemplateTable';
+import Tag from '@/ds-components/Tag';
+import { type RequestError } from '@/hooks/use-api';
+
+import usePermissionsTable from './use-permissions-table';
+
+type Props = {
+  application: Application;
+};
+
+function Permissions({ application }: Props) {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+
+  const [isAssignScopesModalOpen, setIsAssignScopesModalOpen] = useState(false);
+  const { parseRowGroup, deletePermission } = usePermissionsTable();
+
+  const { data, error, mutate } = useSWR<ApplicationUserConsentScopesResponse, RequestError>(
+    `api/applications/${application.id}/user-consent-scopes`
+  );
+
+  const isLoading = !data && !error;
```

Comment:
We can get `isLoading` from swr hook directly now, since v2

Reply:
updated